datumaro/plugins/vgg_face2_format.py

```diff
@@ -43,16 +57,29 @@ def _load_items(self, path):
 
         for row in landmarks_table:
             item_id = row['NAME_ID']
-            image_path = osp.join(self._dataset_dir, self._subset,
-                item_id + VggFace2Path.IMAGE_EXT)
-            annotations = []
+            label = osp.dirname(item_id)
+            if 0 < len(label):
+                item_id = osp.basename(item_id)
+
+            points = None
             if len([p for p in row if row[p] == '']) == 0 and len(row) == 11:
-                annotations.append(Points(
-                    [float(row[p]) for p in row if p != 'NAME_ID']))
-            if item_id in items and 0 < len(annotations):
+                points = Points(
+                    [eval(row[p]) for p in row if p != 'NAME_ID'])
```

Comment:
`eval()` here is a security breach. Why is it used here?

`Points` can have a label too.

`Points` also need to be in a group with `Bbox`, and the `Bbox`, being a leader of the group, should have the same `id` as the `group`. `Points` and other additional descriptors of the object may, or may not have the same `id`.
```
Bbox(id=n, group=n)
Points(group=n)
```

Reply:
`eval()` is needed for `Bbox`. It is necessary that from `str` `int` be cast to `int`, `float` to `float`. From `Points` `eval()` I need to be replaced with `float`.

We can go through folders and create `Label` (if `save_images is true`). Then go through the .csv files and add `Bbox` and `Points`, combining into groups (without labels inside them).

Reply:
> It is necessary that from str int be cast to int, float to float.

Why do we need such a cast for coordinates? All the coordinates are supposed to be `float`.


> We can go through folders ...

I think, it should be done vise-versa - we parse boxes and points, and finish, or parse labels, if there is no boxes. I don't think we need to have both labels and boxes simultaneously in a group, it looks like overkill here. Preferring boxes over labels will simplify exporting to other formats.